### PR TITLE
DOC: Clarify requirements for running on GPU and supported configurations

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

This PR:
* Updates the information about which configurations are supported on each distribution channel (latest conda-forge release has DPC and SPMD modules).
* Updates the list of supported python and sklearn versions to reflect the current situation in the main branch.
* Updates the description of GPU support logic.
* Updates the requirements on GPU software that is needed.
* Improves the examples by adding direct links to DPCTL docs, more code snippets, fixing typos, etc.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.

**Performance**

Not applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,8 +45,8 @@ The software acceleration is achieved with vector instructions, AI hardware-spec
 
 With Intel(R) Extension for Scikit-learn, you can:
 
-* Speed up training and inference by up to 100x with the equivalent mathematical accuracy
-* Benefit from performance improvements across different Intel(R) hardware configurations
+* Speed up training and inference by up to 100x with equivalent mathematical accuracy
+* Benefit from performance improvements across different Intel(R) hardware configurations, including GPUs (discrete/integrated) and FPGAs
```
Comment (napetrov) [inline README.md:49]: FPGAs are not supported

File: doc/sources/oneapi-gpu.rst [hunk 2]
```diff
@@ -18,17 +18,18 @@
 oneAPI and GPU support in |intelex|
 ##############################################################
 
-|intelex| supports oneAPI concepts, which
-means that algorithms can be executed on different devices: CPUs and GPUs.
-This is done via integration with
-`dpctl <https://intelpython.github.io/dpctl/latest/index.html>`_ package that
-implements core oneAPI concepts like queues and devices.
+|intelex| can execute computations on different devices (CPUs, GPUs, FPGAs) through the SYCL framework in oneAPI.
```
Comment (napetrov) [inline doc/sources/oneapi-gpu.rst:21]:
```suggestion
|intelex| can execute computations on different devices (CPUs, GPUs) through the SYCL framework in oneAPI.
```

File: README.md (hunk 1 above)
Comment (david-cortes-intel) [reply README.md]: Thanks, removed it and clarified that the supported devices comes from what oneMath can support.

File: doc/sources/oneapi-gpu.rst (hunk 2 above)
Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst]: Removed.

File: README.md [hunk 3]
```diff
@@ -45,8 +45,8 @@ The software acceleration is achieved with vector instructions, AI hardware-spec
 
 With Intel(R) Extension for Scikit-learn, you can:
 
-* Speed up training and inference by up to 100x with the equivalent mathematical accuracy
-* Benefit from performance improvements across different Intel(R) hardware configurations
+* Speed up training and inference by up to 100x with equivalent mathematical accuracy
+* Benefit from performance improvements across different Intel(R) hardware configurations, including GPUs
```
Comment (ethanglaser) [inline README.md:49]: maybe worth mentioning spmd option as well, something like "hardware, including GPU and multi-GPU configurations"

File: doc/sources/distributed-mode.rst [hunk 4]
```diff
@@ -12,39 +12,85 @@
...
+
+  - Conda-Forge::
+
+      conda install -c conda-forge impi_rt
+
+  - PyPI (not recommended, might require setting additional environment variables)::
+
+      pip install impi_rt
+
+  Using other MPI backends (e.g. OpenMPI) requires building |intelex| from source with that backend.
```
Comment (ethanglaser) [inline doc/sources/distributed-mode.rst:50]: From my experience this is not required - I have used mpich and intel mpi interchangeably from the same sklearnex build.

File: doc/sources/quick-start.rst [hunk 5]
```diff
@@ -191,125 +190,79 @@ To install |intelex|, run:
...
-           - [CPU]
-           - [CPU]
+         * - Operating systems
+           - Windows*, Linux*
+         * - Python versions
+           - 3.9, 3.10, 3.11, 3.12, 3.13
+         * - Devices
+           - CPU, GPU
+         * - Modes
+           - Single, SPMD
```
Comment (ethanglaser) [inline doc/sources/quick-start.rst:254]: Can you share a screenshot of the updated rendering of this table?

**Review by ethanglaser — COMMENTED** [review]
Very nice work on this, the docs are much improved

File: doc/sources/distributed-mode.rst (hunk 4 above)
Comment (david-cortes-intel) [reply doc/sources/distributed-mode.rst:50]: That's probably because IMPI is based off MPICH. Does the same build work with OpenMPI for you?

File: doc/sources/quick-start.rst (hunk 5 above)
Comment (david-cortes-intel) [reply doc/sources/quick-start.rst:254]: Screenshot:
image

File: README.md (hunk 3 above)
Comment (david-cortes-intel) [reply README.md]: Good idea, added.

File: doc/sources/quick-start.rst [hunk 6]
```diff
@@ -191,125 +190,79 @@ To install |intelex|, run:
...
-           - [CPU]
-           - [CPU]
-           - [CPU]
-           - [CPU]
-         * - Windows* OS
-           - [CPU]
-           - [CPU]
-           - [CPU]
-           - [CPU]
+*Note: the main Anaconda channel also provides distributions of scikit-learn-intelex, but it does not provide the latest versions, nor does it provide GPU-enabled builds. It is highly recommended to install it from either Intel's channel or conda-forge instead.*
```
Comment (Alexsandruss) [inline doc/sources/quick-start.rst:213]: Not entirely true: conda-forge linux build contains `dpc` and `spmd_dpc` libs starting from 2025 release. However, GPU support is not tested for it due to HW unavailability in conda-forge-feedstock CI.

**Review by Alexsandruss — APPROVED** [review]

File: doc/sources/quick-start.rst (hunk 6 above)
Comment (david-cortes-intel) [reply doc/sources/quick-start.rst:213]: @Alexsandruss Thanks for the clarification.

But does the build from conda-forge differ from the build from intel-conda in some way that would affect GPU support? As I understand it, both of them are built with the exact same build-time dependencies (compiler, headers, TBB, MKL, MPI) in the exact same versions.

File: doc/sources/distributed-mode.rst (hunk 4 above)
Comment (ethanglaser) [reply doc/sources/distributed-mode.rst:50]: I have not tried with openMPI, but i'll take your word for it

**Review by ethanglaser — APPROVED** [review]

File: doc/sources/quick-start.rst (hunk 6 above)
Comment (Alexsandruss) [reply doc/sources/quick-start.rst:213]: > @Alexsandruss Thanks for the clarification.
> 
> But does the build from conda-forge differ from the build from intel-conda in some way that would affect GPU support? As I understand it, both of them are built with the exact same build-time dependencies (compiler, headers, TBB, MKL, MPI) in the exact same versions.

Toolchains are different: `intel` conda channel used internal, `conda-forge` - public.

Comment (david-cortes-intel) [reply doc/sources/quick-start.rst:213]: But then again: what differences are there in those toolchains that could affect GPU support, if they are at the same version numbers?

**Review by maria-Petrova — APPROVED** [review]
LGTM

File: doc/sources/quick-start.rst (hunk 6 above)
Comment (Alexsandruss) [reply doc/sources/quick-start.rst:213]: It didn't look at it since both work fine anyway.
